src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -51,43 +49,33 @@ internal set
         /// Indicates whether pre-generated serialization logic for types in the context
         /// is compatible with the run time specified <see cref="JsonSerializerOptions"/>.
         /// </summary>
-        internal bool CanUseSerializationLogic
+        internal bool CanUseFastPathSerializationLogic(JsonSerializerOptions options)
```

Comment:
Changes here and in `JsonSerializerWrapper` have been picked up from https://github.com/dotnet/runtime/pull/79828#discussion_r1052520019, since these tests also depend on them.